src/renderer/routes/test_sequencer_panel/components/data-table/SequenceTable.tsx

```diff
@@ -255,8 +258,82 @@ export function SequenceTable() {
     );
   };
 
+  const onRenameSequence = (idx: number) => {
+    const sequence = sequences[idx];
+    renameForIdx.current = idx;
+    setRenameTarget(sequence.project.name);
+    setInitialName(sequence.project.name);
+    setIsRenameNameModalOpen(true);
+  };
+
+  const handleRenameSequence = (newName: string) => {
+    // make sure the new name is unique
+    if (sequences.some((seq) => seq.project.name === newName)) {
+      toast.error("Sequence name must be unique");
+      return;
+    }
+    setSequences(
+      [...sequences].map((seq, idx) => {
+        if (idx === renameForIdx.current) {
+          return {
+            ...seq,
+            project: { ...seq.project, name: newName },
+            testSequenceUnsaved: true,
+          };
+        }
+        return seq;
+      }),
+    );
+    setIsRenameNameModalOpen(false);
+  };
+  const [isRenameNameModalOpen, setIsRenameNameModalOpen] = useState(false);
+  const [isRenameDescModalOpen, setIsRenameDescModalOpen] = useState(false);
+  const [renameTarget, setRenameTarget] = useState("");
+  const [initialName, setInitialName] = useState("");
+  const renameForIdx = useRef(-1);
+
+  const onRenameDescription = (idx: number) => {
+    const sequence = sequences[idx];
+    renameForIdx.current = idx;
+    setRenameTarget(sequence.project.name);
+    setInitialName(sequence.project.description);
+    setIsRenameDescModalOpen(true);
+  };
+
+  const handleRenameDescription = (newDescription: string) => {
+    setSequences(
+      [...sequences].map((seq, idx) => {
+        if (idx === renameForIdx.current) {
+          return {
+            ...seq,
+            project: { ...seq.project, description: newDescription },
+            testSequenceUnsaved: true,
+          };
+        }
+        return seq;
+      }),
+    );
```

Comment:
consider using immer whenever you need to modify nested properties, it's a lot more readable
```typescript
setSequences(produce(sequences, (draft) => {
  const seq = draft[renameForIdx.current];
  seq.project.description = newDescription;
  seq.testSequenceUnsaved = true;
}))
```

Reply:
Oh that's wayyyy better